text/0000-style-properties.md

```diff
@@ -0,0 +1,242 @@
+
+In unsupported browsers, there is a chance it would break some layouts (though setting `width: 100%; height: 100%` would fix many of those). Those browsers generally don't support custom properties anyway, so this should probably be considered a modern-browser-only feature.
+
+🐃 It *would* be possible for someone to accidentally target the `<div>`, so it may be preferable to use a made-up element name instead (`<styles>`?). In SVG, there might not be any choice but to use a `<g>`. In 99.9% of cases it wouldn't matter in the least, but it *would* need to be documented.
+
+
+### Global theming
+
+Because this approach uses custom properties, it becomes straightforward to set theme styles globally, or for a large subtree of the app:
+
+```css
+/* global.css */
+html {
+  --rail-color: black;
+  --track-color: red;
+}
+```
+
+```html
+<!-- App.svelte -->
+<div>
+  <OtherStuff/>
+</div>
+
+<style>
+  div {
+    --rail-color: black;
+    --track-color: red;
+  }
+</style>
+```
+
+A consumer of the component can override them...
+
+```html
+<Slider --track-color="goldenrod"/>
+```
+
+...but ultimately the component itself has control over what is exposed, and can specify its own fallback values using normal CSS custom property syntax:
+
+```html
+<!-- Slider.svelte -->
+<style>
+  .potato-slider-rail {
+    background-color: var(--rail-color, var(--potato-theme-color, 'purple'));
+  }
+</style>
+```
+
+
+## How we teach this
+
+This ought to be straightforward to teach, at least to people already familiar with custom properties. Terminology-wise, it probably makes sense to refer to 'style props' to distinguish them from regular component props.
+
+It would require a new tutorial chapter and updated documentation.
+
+
+## Drawbacks
+
+*Technically* this would be a breaking change, since `--foo` is currently passed down as a regular prop (albeit only accessible via `$$props['--foo']`). I feel pretty confident saying this isn't a real world concern.
+
+There are some more salient drawbacks:
+
+* It relies on something that is inherently global. Different components might 'claim' a given property name. While it's possible to differentiate them at the subtree level, it's not possible to do so globally.
+* The compiler would need to generate extra code for every component (for applying received style properties, and for passing them on to top-level child components), regardless of whether they were actually used, and the helper would be included in everyone's app. (This *could* be avoided with some form of whole-app optimisation.)
```

Comment:
I think this bullet point can be removed now with the new `display: contents` thingy.